### PR TITLE
[UNR-2978][MS] Added warning on deploying SimPlayers without it being built on your machine.

### PR DESCRIPTION
**Pull request by MatthewSandfordImprobable**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

I will be modifying BuildWorker.bat in our projects to remove the building of SimPlayers. This reflects what we say in the docs (that BuildWorker does not build SimPlayers).

https://github.com/spatialos/UnrealGDKExampleProject/pull/104
https://github.com/spatialos/UnrealGDKTestGyms/pull/42

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKSimulatedPlayerDeployment.cpp [hunk 1]
```diff
@@ -564,6 +566,20 @@ FReply SSpatialGDKSimulatedPlayerDeployment::OnLaunchClicked()
 		return FReply::Handled();
 	}
 
+	if (SpatialGDKSettings->IsSimulatedPlayersEnabled())
+	{
+		IPlatformFile& PlatformFile = FPlatformFileManager::Get().GetPlatformFile();
+		FString BuiltWorkerFolder = GetDefault<USpatialGDKEditorSettings>()->GetBuiltWorkerFolder();
+		FString BuiltSimPlayersName = TEXT("UnrealSimulatedPlayer@Linux.zip");
```
Comment (mironec) [inline SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKSimulatedPlayerDeployment.cpp]: Slightly worried whether we have this name used anywhere else? In the docs?

Comment (MatthewSandfordImprobable) [reply SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKSimulatedPlayerDeployment.cpp]: And that we should have this stored somewhere globally accessible and only have one place that sets it?

What is the concern with docs?

Comment (mironec) [reply SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKSimulatedPlayerDeployment.cpp]: Yes, to the first part.
I meant, how will people know that to build simplayers, they need to have this exact assembly name?

Comment (MatthewSandfordImprobable) [reply SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKSimulatedPlayerDeployment.cpp]: Testing building SimPlayers manually it seems UnrealSimulatedPlayer@Linux is alwas the ouput of BuildWorker.bat. In otherwords it doesnt matter what the customer will put in as a name for the SimPlayer build. We do have docs on this:

image

Comment (MatthewSandfordImprobable) [reply SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKSimulatedPlayerDeployment.cpp]: As long as they follow the docs they output zipped file should be "UnrealSimulatedPlayer@Linux.zip"

Comment (MatthewSandfordImprobable) [reply SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKSimulatedPlayerDeployment.cpp]: Looking in the code base I can not find another use of UnrealSimulatedPlayer@Linux

Comment (mironec) [reply SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKSimulatedPlayerDeployment.cpp]: Ah, okay, <YourProject>SimulatedPlayer dictates the name of the assembly and it's in docs, all good.

**Pull request merged**